### ISSUE TITLE
compiler warning

### ISSUE THREAD

**Issue by sdirkse67** [ERGO-Code/HiGHS#1571]
I recently updated my GCC to 13.21.  And I pulled the latest from master.  And I get this warning message:

```
[ 99%] Building CXX object app/CMakeFiles/highs-bin.dir/RunHighs.cpp.o
In file included from /home/sdirkse/github/HiGHS/master/app/RunHighs.cpp:16:
In file included from /home/sdirkse/github/HiGHS/master/src/lp_data/HighsRuntime     Options.h:15:
/home/sdirkse/github/HiGHS/master/app/cxxopts.hpp:488:36: warning: overflow in e     xpression; result is -2147483648 with type 'int' [-Winteger-overflow]
  488 |             if (u > static_cast<U>(-std::numeric_limits<T>::min()))
      |                                    ^
/home/sdirkse/github/HiGHS/master/app/cxxopts.hpp:488:36: warning: overflow in e     xpression; result is -9223372036854775808 with type 'long' [-Winteger-overflow]
2 warnings generated.
[100%] Linking CXX executable ../bin/highs
[100%] Built target highs-bin
```

Taking -std::numeric_limits<int>::min() doesn't look right to me.  What constant results from -MIN_INT ??

**Comment by jajhall** [comment]
I don't know. Plus this is in `cxxopts.hpp`, which isn't our code. Maybe there's a CXX forum in which this can be raised?

**Comment by Coloquinte** [comment]
This was changed on [recent cxxopt](https://github.com/jarro2783/cxxopts/blob/master/include/cxxopts.hpp).

At the bit level MIN_INT = -MIN_INT (usually), so it seems they just had to remove the '-'

**Comment by jajhall** [comment]
You're a star!

**Comment by jajhall** [comment]
Unfortunately the latest version yields

> /home/jajhall/HiGHS/app/cxxopts.hpp:2288:5: warning: use of the 'fallthrough' attribute is a C++17 extension [-Wc++17-attribute-extensions]
    CXXOPTS_FALLTHROUGH;
    ^
/home/jajhall/HiGHS/app/cxxopts.hpp:80:35: note: expanded from macro 'CXXOPTS_FALLTHROUGH'
    #define CXXOPTS_FALLTHROUGH [[fallthrough]]
                                  ^
In file included from /home/jajhall/HiGHS/app/RunHighs.cpp:16:
/home/jajhall/HiGHS/src/lp_data/HighsRuntimeOptions.h:248:27: error: no type named 'OptionException' in namespace 'cxxopts'
  } catch (const cxxopts::OptionException& e) {
                 ~~~~~~~~~^
1 warning and 1 error generated.
make[2]: *** [app/CMakeFiles/highs.dir/build.make:76: app/CMakeFiles/highs.dir/RunHighs.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:896: app/CMakeFiles/highs.dir/all] Error 2

So I'll flip the sign and wait for @galabovaa to see what's the most up-to-date version we can use

**Comment by jajhall** [comment]
Closed by #1575
